src/java.base/share/classes/java/lang/constant/MethodTypeDesc.java

```diff
@@ -196,11 +202,11 @@ static MethodTypeDesc of(ClassDesc returnDesc, ClassDesc... paramDescs) {
      * @jvms 4.3.3 Method Descriptors
      */
     default String descriptorString() {
-        return String.format("(%s)%s",
-                             Stream.of(parameterArray())
-                                   .map(ClassDesc::descriptorString)
-                                   .collect(Collectors.joining()),
-                             returnType().descriptorString());
+        var sj = new StringJoiner("", "(", ")" + returnType().descriptorString());
```

Comment:
`MethodTypeDescImpl` is the only class implementing `MethodTypeDesc` and implements `descriptorString()`.    This default implementation is not needed.   Should the implementation be moved to `MethodTypeDescImpl::descriptorString`?